ansible/roles/cni/defaults/main.yml

```diff
@@ -1,2 +1,8 @@
 ---
 cni_plugin: calico
+bgp_peer_address: 192.168.0.1
+bgp_peer_asn: 64512
+cilium_helm_version: 1.8.3
+cilium_image_version: v1.8.3
+k8s_service_host: "{{ hostvars[groups['masters'][0]]['ansible_host'] }}"
```

Comment:
@RobReus -- This really should be VIP IP if it's presented, I'm at the edge of my Ansible skills.

Reply:
when using `keepalived` this is associated with [keepalived_vip](https://github.com/raspbernetes/k8s-cluster-installation/blob/49dc67dd974f494e22861e38f8d37088d9463778/ansible/group_vars/masters.yml#L3), but i don't know how running cilium affects running keepalived?

Reply:
theoretically it should be outside of Cilium as it's a host level config, right?

However, this would need to be an either or I think; if VIP present use that, otherwise use the master IP?

Reply:
yeah i think you're right. it should be an either/or situation. i'm not sure if something could be achieved with jinja logic, but my assumption is that it would be possible by using some jinja operators on the template, in addition to some sanity checking for each var.